packages/mermaid/src/diagrams/xychart/xychartDb.ts

```diff
@@ -0,0 +1,231 @@
+import type { Selection } from 'd3-selection';
+import {
+  clear as commonClear,
+  getAccDescription,
+  getAccTitle,
+  getDiagramTitle,
+  setAccDescription,
+  setAccTitle,
+  setDiagramTitle,
+} from '../common/commonDb.js';
+import * as configApi from '../../config.js';
+import defaultConfig from '../../defaultConfig.js';
+import { getThemeVariables } from '../../themes/theme-default.js';
+import { cleanAndMerge } from '../../utils.js';
+import { sanitizeText } from '../common/common.js';
+import { XYChartBuilder } from './chartBuilder/index.js';
+import type {
+  DrawableElem,
+  SimplePlotDataType,
+  XYChartConfig,
+  XYChartData,
+  XYChartThemeConfig,
+} from './chartBuilder/interfaces.js';
+import { isBandAxisData, isLinearAxisData } from './chartBuilder/interfaces.js';
+
+export type SVGGType = Selection<SVGGElement, unknown, Element | null, unknown>;
+
+let plotIndex = 0;
+
+let tmpSVGGElem: SVGGType;
+
+let xyChartConfig: XYChartConfig = getChartDefaultConfig();
+let xyChartThemeConfig: XYChartThemeConfig = getChartDefaultThemeConfig();
+let xyChartData: XYChartData = getChartDefalutData();
+let plotColorPalette = xyChartThemeConfig.plotColorPalette.split(',').map((color) => color.trim());
+let hasSetXAxis = false;
+let hasSetYAxis = false;
+
+interface NormalTextType {
+  type: 'text';
+  text: string;
+}
+
+function getChartDefaultThemeConfig(): XYChartThemeConfig {
+  const defaultThemeVariables = getThemeVariables();
+  const config = configApi.getConfig();
+  return cleanAndMerge(defaultThemeVariables.xyChart, config.themeVariables.xyChart);
+}
+function getChartDefaultConfig(): XYChartConfig {
+  const config = configApi.getConfig();
+  return cleanAndMerge<XYChartConfig>(
+    defaultConfig.xyChart as XYChartConfig,
+    config.xyChart as XYChartConfig
```

Comment:
Aren't these properties typed already? Ideally, we shouldn't have to cast these explicitly.

Reply:
We need to over here as the `defaultConfig.xyChart` has a type `RequiredObjectDeep<XYChartConfig>`
and the `config.xyChart` is autogenerated so the xAxis type does not match the interface defined